### DISCUSSION TITLE
支持EN换班和肉鸽需要做什么呢？

### DISCUSSION THREAD

**Discussion by cookie223** [MaaAssistantArknights/MaaAssistantArknights#1212] [General]
只是提供翻译的话可以帮忙……

**Comment by BlackNorton** [comment]
希望EN尽快支持自动肉鸽刷蜡烛和投资，拜托了各位程序员大佬们

**Comment by amazing-background** [comment]
同样可以帮忙

**Comment by MistEO** [comment]
需要帮忙截图，大量的截图。。

`resource/template` 里所有图片中出现了 汉字/字母/数字 的图片全都要（因为外服的字体不一样）

**Comment by MistEO** [comment]
简单说下方法，各个服都是如此操作

## 修改模板图片

1. 模拟器设置为 1280*720 320 dpi，使用模拟器自带的截图工具 或者 adb 截取原图
3. 参照 `resource/template` 中的截图，同样的大小和位置（不要求严格精确，差不多就行）
4. 如果图片有涂黑的需要手动用 ps 或者画图板之类的同样的涂黑一下
5. 裁剪完的图片放到 `global` 文件夹中对应的位置
6. 重新打开 maa 测试是否生效

tips: 仓库目录 `tools/CropRoi` 有一个用来裁剪的小工具；原图放在 `src` 文件夹里；鼠标滑动选区，`s` 键保存，`q` 键退出；保存完的在 `dst` 文件夹里，有兴趣的话可以研究下怎么用

## 修改 tasks.json

1. 参照国服 `resource/tasks.json` 中所有 `text` 字段，涉及到任务相关的，在 `global` 对应的 `tasks.json` 中原样复制这个 task, 将 `text` 字段内容修改为对应外服的文字
2. 重新打开 maa 测试是否生效

**Reply by BlackNorton** [reply]
截图了EN的肉鸽，打包传倒这里了 https://we.tl/t-rpaOyLU83X
自己试着导入运行，但是在开始界面就会退出。

**Reply by MistEO** [reply]
来点日志看看（可以直接提个 issue），顺便问下有没有小号可以提供的，我这里美服的测试账号的进度还没发打肉鸽 orz

**Reply by BlackNorton** [reply]
7月19日的全部Log都截出来放这里了 https://we.tl/t-XPIboOjgLa
应该有三四次尝试的记录
账号的话，这个抱歉，没有小号，国际服就是我主号，不大方便借

**Reply by cookie223** [reply]
```
[2022-07-19 18:05:47.433][TRC][Px4724][Txeec] match_templ | Roguelike1Abandon.png score: 0.616015 rect: [ 1128, 294, 71, 76 ] roi: [ 1076, 243, 177, 179 ]
[2022-07-19 18:05:47.433][TRC][Px4724][Txeec] asst::ProcessTask::_run | leave, 109 ms
```
这个看起来是match成了Abandon然后试图leave（猜测这个function是点abandon？）
而且不断重复，猜测是没点上。。不知道是不是控制的问题
其他已经适配好的功能可以用吗？比如visit friends?

**Reply by cookie223** [reply]
我在debug……你的dimension应该是不太对……好几个识别不出来了，Roguelike1QuickFormation这个一看就比较长……

**Comment by MistEO** [comment]
日服的肉鸽 @natt3 已经在帮忙适配啦，说一下避免撞车了_(:з」∠)_

**Reply by MistEO** [reply]
https://github.com/MaaAssistantArknights/MaaAssistantArknights/pull/1233 来了！

**Comment by cookie223** [comment]
找了个朋友在做EN的Operator names了……screenshot工作量有点大……

**Comment by cookie223** [comment]
screenshots still needed: (assuming the ones without texts are fine)
Rouguelike1Continue
Roguelike1GetDrop2
Roguelike1GetDropSelect
Roguelike1GetDropSelectReward
Roguelike1LastReward3
Roguelike1RecruitCloseGuide
Roguelike1RecruitConfirm
Roguelike1StageDreadfulFoeEnter
Roguelike1StageSafeHouseEnter
Roguelike1TeamConfirm

**Reply by BlackNorton** [reply]
你看看我之前截图的en肉鸽包里有没有可以用的图
https://we.tl/t-rpaOyLU83X
如果不适配说一下我再去截

**Reply by cookie223** [reply]
> 你看看我之前截图的en肉鸽包里有没有可以用的图 https://we.tl/t-rpaOyLU83X 如果不适配说一下我再去截

昨天试图用了一些你的图……但是并不对，大概是分辨率不对
需要模拟器 1280*720 320 dpi

**Reply by BlackNorton** [reply]
奇怪我确实是按那个配置来的
不过如果不小心拉动了模拟器窗口的边框是不是会影响？
我截图的时候可能不小心碰到过

**Reply by cookie223** [reply]
用模拟器的screenshot 比如
![image](https://user-images.githubusercontent.com/5301377/180011972-dd2e676f-7abd-47f5-844d-bb9c3d24bd84.png)

这样截出来整个图一定是1280*720，再裁。

**Comment by zzyyyl** [comment]
现在日服发现了有一些模板图片位置不同的问题，如果 EN 也有记得说一下

**Reply by cookie223** [reply]
在游戏内的位置不同？不玩国服估计注意不到……
